CHANGES: update changelog for 24.0 release (#1385)
labgrid-project/labgrid: CHANGES.rst

- The tool lookup function has been extended to return the original name in case
  the path can't be found. This makes specification of the qemu binary to use
  easier.
- The bindings baseclass has been extended, allowing the user to retrieve all
  resources used by a driver.
- Support for STLink V2 was added.
- UBootStrategy was extended with a force() function.
- labgrid was switched from pysnmp to pysnmp-lexstudio.
- Support for Segger J-Link was added.
- Place tags are now exposed by the RemotePlace.
- The sync-places contrib script has gained support for named matches.
- Remote support for YKush Devices was added.
- Support for sigrok DMMs was added.
- Support for Digital Outputs switched via HTTP was added.
- The ``QEMUDriver`` has a new get_qemu_base_args() function which can be used to
  extract the arguments passed to qemu.
- The ``SSHDriver`` has gained support to forward unix sockets.
- The exporter has gained an ``--fqdn`` argument to set the hostname to the
  fully qualified domain name instead of the hostname.
- The ``QEMUDriver`` now has an additional ``disk_opts`` property which can be
  used to pass additional options for the disk directly to QEMU

> - All drivers now inherit a logger from the ``Driver`` baseclass and many

Reviewer: ```suggestion
- All drivers now inherit a logger from the ``Driver`` base class and many
```

  drivers were changed to use this logger.

> - The new ``poe_mib`` backend allows switching of power over ethernet capable

Reviewer: ```suggestion
- The new ``poe_mib`` backend allows switching of power over Ethernet-capable
```

  ports on switches that use the corresponding SNMP MIB.
- The ``RawNetworkInterfaceDriver`` allows the replay and recording of network
  packets on ethernet interfaces.

> - The i.MX93 usb loader ID has been added.

Reviewer: ```suggestion
- The i.MX93 usb loader USB ID has been added.
```

Reviewer 2: Maybe mention the corresponding resource?

  of pip.
- Several tests have gained an importorskip() call to skip them if the
  module is not available.

> - labgrid now uses it's own pyserial fork from pypi since installation from

Reviewer: ```suggestion
- labgrid now uses its own pyserial fork from pypi since installation from
```

  github as an egg is no longer properly supported.
- The build-and-release workflow supports building wheels.
- Fix named SSH lookups in conjunction with an environment file in
  labgrid-client.
- The crossbar virtual-environment now needs to be separate from the labgrid

>   environment, for more information please consult the current documentation.

Reviewer: Could we link to the documentation here?

- The markers now are restricted to patterns which won't match WARN,
  ERROR, INFO and similar log notifiers.
- A race inside the ``SSHDriver`` cleanup has been fixed.

> - The labgrid-client monitor command now outputs the full resource identifier.

Reviewer: ```suggestion
- The ``labgrid-client monitor`` command now outputs the full resource identifier.
```

- Many of the USB loader commands e.g. imx-usb-loader will now print to the
  console when logging is not enabled.

> - An UnboundLocalError inside the atomic_replace code which is used inside the

Reviewer: ```suggestion
- An ``UnboundLocalError`` inside the atomic_replace code which is used inside the
```

  coordinator was fixed.
- Resources of different classes can now have the same name.
- A bug within the pytest logging setup was fixed.
- The ``QemuDriver`` correctly handles the different command lines for virgl
  enablement.
- A bug was fixed where resource names were ignored during lookup of the correct
  power driver.
- ManagedFile was fixed to work with the stat command on Darwin.
- Instead of using a private member on the pytest config, the labgrid plugin now
  uses the pytest config stash.
- The ``ShellDriver`` was fixed to set the correct status attribute.
- The USBNetworkInterface now warns if the interface name is set, as it will be

>   overwritten by the ResourceManager to assign the correct interface name

Reviewer: ```suggestion
  overwritten by the ResourceManager to assign the correct interface name.
```

- Fix sftp option issue in SSH driver that caused sftp to only work once per
  test run.
- ManagedFile NFS detection heuristic now does symlink resolution on the
  local host.

> - XModem support within the shelldriver was fixed by removing the newline from

Reviewer: ```suggestion
- XModem support within the ShellDriver was fixed by removing the newline from
```
